src/Middleware/Spa/SpaProxy/src/SpaProxyLaunchManager.cs

```diff
@@ -203,38 +221,86 @@ private void LaunchDevelopmentProxy()
-                    };
+            var stopScriptInfo = new ProcessStartInfo(
+                "powershell.exe",
+                string.Join(" ", "-NoProfile", "-C", stopScript))
+            {
+                CreateNoWindow = true,
+                WorkingDirectory = Path.Combine(AppContext.BaseDirectory, _options.WorkingDirectory)
+            };
 
-                    var stopProcess = Process.Start(stopScriptInfo);
-                    if (stopProcess == null || stopProcess.HasExited)
-                    {
-                        _logger.LogWarning($"SPA process shutdown script '{stopProcess?.Id}' failed to start. The SPA proxy might" +
-                            $" remain open if the dotnet process is terminated abruptly. Use the operating system command to kill" +
-                            $"the process tree for {_spaProcess.Id}");
-                    }
-                    else
-                    {
-                        _logger.LogDebug($"Watch process '{stopProcess}' started.");
-                    }
-                }
+            var stopProcess = Process.Start(stopScriptInfo);
+            if (stopProcess == null || stopProcess.HasExited)
+            {
+                _logger.LogWarning($"The SPA process shutdown script '{stopProcess?.Id}' failed to start. The SPA proxy might" +
+                    $" remain open if the dotnet process is terminated ungracefully. Use the operating system commands to kill" +
+                    $" the process tree for {_spaProcess!.Id}");
             }
-            catch (Exception exception)
+            else
             {
-                _logger.LogError(exception, $"Failed to launch the SPA development server '{_options.LaunchCommand}'.");
+                _logger.LogDebug($"Watch process '{stopProcess}' started.");
             }
         }
 
+        private void LaunchStopScriptMacOS()
+        {
+            var fileName = Guid.NewGuid().ToString("N") + ".sh";
+            var scriptPath = Path.Combine(AppContext.BaseDirectory, fileName);
+            var stopScript = @$"function list_child_processes(){{
+    local ppid=$1;
+    local current_children=$(pgrep -P $ppid);
+    local local_child;
+    if [ $? -eq 0 ];
+    then
+        for current_child in $current_children
+        do
+          local_child=$current_child;
+          list_child_processes $local_child;
+          echo $local_child;
+        done;
+    else
+      return 0;
+    fi;
+}}
+
+ps {Environment.ProcessId};
+while [ $? -eq 0 ];
+do
+  sleep 1;
+  ps {Environment.ProcessId} > /dev/null;
+done;
+
+for child in $(list_child_processes {_spaProcess!.Id});
+do
+  echo killing $child;
+  kill -s KILL $child;
+done;
+rm {scriptPath};
+";
+            File.WriteAllText(scriptPath, stopScript);
+
+            var stopScriptInfo = new ProcessStartInfo(
+                "/bin/bash",
+                string.Join(" ", scriptPath))
```

Comment:
I'm not sure why there's a `string.Join` here. Isn't it a single argument?

Reply:
there used to be many :), good catch

Reply:
done